src/libsyntax/feature_gate.rs

```diff
@@ -938,6 +938,12 @@ pub const BUILTIN_ATTRIBUTES: &[(&str, AttributeType, AttributeTemplate, Attribu
                                       is just used for rustc unit tests \
                                       and will never be stable",
                                      cfg_fn!(rustc_attrs))),
+    ("rustc_layout", Normal, template!(Word), Gated(Stability::Unstable,
+                                     "rustc_attrs",
+                                     "the `#[rustc_layout]` attribute \
+                                      is just used for rustc unit tests \
+                                      and will never be stable",
+                                     cfg_fn!(rustc_attrs))),
```

Comment:
Why not use the layout printing flag (I still think it's `-Z print-type-sizes`, which we should probably change)?

Reply:
well, first reason is that I didn't know about that =)

but that seems like it will also print too much information (all the layout, even stuff we don't care about, resulting in us needed to update the stderr more often) but also not enough (it won't print in particular the homogeneous aggregate deterination)